frontend/test/pytest/test_gradient.py

```diff
@@ -261,6 +261,14 @@ def f3(x: float):
     assert np.allclose(result[0]["helloworld"], expected[0]["helloworld"])
     assert np.allclose(result[1]["helloworld"], expected[1]["helloworld"])
 
+    @qjit
+    def f4(x: float, y: float, z: float):
+        return 100 * x + 200 * y + 300 * z
+
+    result = qjit(value_and_grad(f4))(0.1, 0.2, 0.3)
+    expected = (140, 100)
```

Comment:
Could separate the test? Also add a test with a qnode and parametrize the single argnum